keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/ResourceValidator.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.keel.plugin
+
+import com.netflix.spinnaker.keel.api.Resource
+import kotlin.reflect.KClass
+
+/**
+ * A resource validator throws a [FailedValidationException] when a resource is invalid
+ */
+interface ResourceValidator<T : Any> {
```

Comment:
I think `T : Any` is redundant (could just use `<T>`)

Reply:
no, because then it's equivalent to `<T: Any?>`